internal/encryption/encryption.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (c) 2023 HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package encryption
+
+import (
+	"github.com/opentofu/opentofu/internal/encryption/config"
+	"github.com/opentofu/opentofu/internal/encryption/registry"
+)
+
+// New creates a new Encryption provider from the given configuration and registry.
+func New(reg registry.Registry, cfg *config.Config) Encryption {
```

Comment:
Tiny tiny tiny tiny tiny tiny nitpick: we should move this New method below the interface